src/data/types.ts

```diff
@@ -0,0 +1,35 @@
+import { Static, Type } from '@sinclair/typebox';
+
+export const PineconeConfigurationSchema = Type.Object(
+  {
+    environment: Type.String({ minLength: 1 }),
+    apiKey: Type.String({ minLength: 1 }),
+    projectId: Type.Optional(Type.String({ minLength: 1 })),
+  },
+  { additionalProperties: false }
+);
+export type PineconeConfiguration = Static<typeof PineconeConfigurationSchema>;
+
+export const RecordIdSchema = Type.String({ minLength: 1 });
+export const RecordValuesSchema = Type.Array(Type.Number());
+export const RecordSparseValuesSchema = Type.Object(
+  {
+    indices: Type.Array(Type.Integer()),
+    values: Type.Array(Type.Number()),
+  },
+  { additionalProperties: false }
+);
+export const PineconeRecordSchema = Type.Object(
+  {
+    id: RecordIdSchema,
+    values: RecordValuesSchema,
+    sparseValues: Type.Optional(RecordSparseValuesSchema),
+    metadata: Type.Optional(Type.Object({}, { additionalProperties: true })),
```

Comment:
**question:** Thoughts on `metadata` having it's own type along with the other fields?